drivers/iio/dac/ad5770r.c

```
@@ -0,0 +1,715 @@
+		ad5770r_min_max_table[AD5770R_CH0].min = -60;
+		ad5770r_min_max_table[AD5770R_CH0].max = 0;
+	} else if (st->output_mode[AD5770R_CH0].output_range_mode == 0x02) {
+		ad5770r_min_max_table[AD5770R_CH0].min = -60;
+		ad5770r_min_max_table[AD5770R_CH0].max = 300;
+	}
+
+	/* Low headroom */
+	if (st->output_mode[AD5770R_CH1].output_range_mode == 0x01) {
+		ad5770r_min_max_table[AD5770R_CH1].min = 0;
+		ad5770r_min_max_table[AD5770R_CH1].max = 140;
+	/* Low noise and PSRR */
+	} else if (st->output_mode[AD5770R_CH1].output_range_mode == 0x02) {
+		ad5770r_min_max_table[AD5770R_CH1].min = 0;
+		ad5770r_min_max_table[AD5770R_CH1].max = 140;
+	} else if (st->output_mode[AD5770R_CH1].output_range_mode == 0x03) {
+		ad5770r_min_max_table[AD5770R_CH1].min = 0;
+		ad5770r_min_max_table[AD5770R_CH1].max = 250;
+	}
+
+	if (st->output_mode[AD5770R_CH2].output_range_mode == 0x00) {
+		ad5770r_min_max_table[AD5770R_CH2].min = 0;
+		ad5770r_min_max_table[AD5770R_CH2].max = 55;
+	} else if (st->output_mode[AD5770R_CH2].output_range_mode == 0x01) {
+		ad5770r_min_max_table[AD5770R_CH2].min = 0;
+		ad5770r_min_max_table[AD5770R_CH2].max = 150;
+	}
+
+	for (i = AD5770R_CH2; i < AD5770R_MAX_CHANNELS; i++)
+		if (st->output_mode[i].output_range_mode == 0x00) {
+			ad5770r_min_max_table[i].min = 0;
+			ad5770r_min_max_table[i].max = 45;
+		} else if (st->output_mode[i].output_range_mode == 0x01) {
+			ad5770r_min_max_table[i].min = 0;
+			ad5770r_min_max_table[i].max = 100;
+		}
+};
+
+static void ad5770r_parse_dt(struct ad5770r_state *st)
+{
+	unsigned int i;
+	u8 tmparray[8], tmp;
+
+	/* Default range for CH1 */
+	st->output_mode[AD5770R_CH1].output_range_mode = 2;
+	if (!device_property_read_u8_array(&st->spi->dev, "adi,output",
+					    tmparray, 6))
+		for (i = 0; i < AD5770R_MAX_CHANNELS; i++)
+			st->output_mode[i].output_range_mode = tmparray[i];
+
+	ad5770r_update_range(st);
+
+	if (!device_property_read_u8_array(&st->spi->dev, "adi,alarm",
```

general comment/question for these device-tree properties: do we need to throw an error for some of them ? or can all of the be optional ?